File: app/src/main/java/com/lttrung/dormitory/database/repositories/UserRepositories.java
package com.lttrung.dormitory.database.repositories;

import java.lang.reflect.Type;

import javax.inject.Singleton;

import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Single;

@Singleton
public interface UserRepositories {
    Single<Type> fetchProfile();
    Completable changePassword();
    Completable editProfile();
}

File: app/src/main/java/com/lttrung/dormitory/database/repositories/UserRepositories.kt
package com.lttrung.dormitory.database.repositories

import com.lttrung.dormitory.database.data.local.UserLocal
import com.lttrung.dormitory.database.data.network.UserNetwork
import com.lttrung.dormitory.database.data.network.models.FetchRoomContractResponse
import com.lttrung.dormitory.database.data.network.models.UserProfile
import io.reactivex.rxjava3.core.Single
import javax.inject.Singleton

@Singleton
interface UserRepositories {
    val local: UserLocal
    val network: UserNetwork
    fun fetchProfile(): Single<UserProfile>
    fun changePassword(oldPassword: String, newPassword: String): Single<Boolean>
    fun getRoomContract(): Single<FetchRoomContractResponse>
    fun registerRoom(roomId: Int): Single<Boolean>
    fun logout()
}